File: E_Commercial_Mobile_FrontEnd/models/Promotion.ts
class PromotionModel {
    id: number;
    code: string;
    description: string;
    discount: number;
    active: boolean;
    startDate: Date;
    endDate: Date;

    constructor(id: number, code: string, description: string, discount: number, active: boolean, startDate: Date, endDate: Date) {
        this.id = id;
        this.code = code;
        this.description = description;
        this.discount = discount;
        this.active = active;
        this.startDate = startDate;
        this.endDate = endDate;
    }
}

export default PromotionModel;

File: E_Commercial_Mobile_Backend/src/main/java/vn/edu/iuh/fit/entity/Promotion.java
/*
 * @ (#) Discount.java       1.0     19/11/2024
 *
 * Copyright (c) 2024 IUH. All rights reserved.
 */

package vn.edu.iuh.fit.entity;
/*
 * @description:
 * @author: Nguyen Thanh Nhut
 * @date: 19/11/2024
 * @version:    1.0
 */

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Entity
@Table(name = "promotion")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Promotion {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String code;
    private String description;
    private int discount;
    private boolean active;
    private LocalDate startDate;
    private LocalDate endDate;

}
